File: 10000/800/10844_쉬운_계단_수/BJ_10844_쉬운_계단_수.py
# https://www.acmicpc.net/problem/10844

N = int(input())

cache = [[0] * 10 for _ in range(N)]
cache[0] = [1] * 10

for length in range(1, N):
    for i in range(0, 10):
        if 0 < i:
            cache[length][i] += cache[length - 1][i - 1]
            cache[length][i] %= 1_000_000_000
        if i < 9:
            cache[length][i] += cache[length - 1][i + 1]
            cache[length][i] %= 1_000_000_000

ans = 0
for i in range(1, 10):
    ans += cache[N - 1][i]
    ans %= 1_000_000_000

print(ans)

File: 10000/800/10844_쉬운_계단_수/BJ_10844_쉬운_계단_수.java
// https://www.acmicpc.net/problem/10844

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;

public class BJ_10844_쉬운_계단_수 {
    private static final int M = 1_000_000_000;

    public static void main(String[] args) throws IOException {
        final var stdout = new BufferedWriter(new OutputStreamWriter(System.out));
        final var stdin = new BufferedReader(new InputStreamReader(System.in));

        final var N = Integer.parseInt(stdin.readLine());
        final var dp = new long[N][10];
        Arrays.fill(dp[0], 1);
        dp[0][0] = 0;
        for (var i = 1; i < N; ++i) {
            for (var j = 0; j < 10; ++j) {
                dp[i][j] += (1 <= j ? dp[i - 1][j - 1] : 0);
                dp[i][j] += (j <= 8 ? dp[i - 1][j + 1] : 0);
                dp[i][j] %= M;
            }
        }

        stdout.write(Arrays.stream(dp[N - 1]).sum() % M + "\n");

        stdout.flush();
        stdout.close();
        stdin.close();
    }
}
